file.go

```diff
@@ -917,47 +917,81 @@ func (f *File) FlattenBatches() (*File, error) {
 		return nil, err
 	}
-
 	of := NewFile()
 
-	// Slice of BatchHeaders
-	sbh := make([]string, 0)
-	for _, b := range f.Batches {
-		bh := b.GetHeader().String()
-		sbh = append(sbh, bh[:87])
-	}
-
-	// Remove duplicate BatchHeader entries
-	sbh = removeDuplicateBatchHeaders(sbh)
-
-	// Add new batches for flattened file
-	for _, record := range sbh {
-
-		bh := flattenBatchHeaderParse(record)
+	if f.Batches != nil {
+		// Slice of BatchHeaders
+		sbh := make([]string, 0)
+		for _, b := range f.Batches {
+			bh := b.GetHeader().String()
+			sbh = append(sbh, bh[:87])
+		}
+		// Remove duplicate BatchHeader entries
+		sbh = removeDuplicateBatchHeaders(sbh)
+		// Add new batches for flattened file
+		for _, record := range sbh {
 
-		b, _ := NewBatch(bh)
-		of.AddBatch(b)
-	}
+			bh := flattenBatchHeaderParse(record)
 
-	for _, batch := range f.Batches {
-		fbh := batch.GetHeader().String()[:87]
-		// Add entries for batches
-		for i, ofBatch := range of.Batches {
-			if strings.EqualFold(fbh, ofBatch.GetHeader().String()[:87]) {
-				entries := batch.GetEntries()
-				for _, entry := range entries {
-					of.Batches[i].AddEntry(entry)
+			b, _ := NewBatch(bh)
+			of.AddBatch(b)
+		}
+		for _, batch := range f.Batches {
+			fbh := batch.GetHeader().String()[:87]
+			// Add entries for batches
+			for i, ofBatch := range of.Batches {
+				if strings.EqualFold(fbh, ofBatch.GetHeader().String()[:87]) {
+					if ofBatch.GetHeader().StandardEntryClassCode == "ADV" {
+						entries := batch.GetADVEntries()
+						for _, advEntry := range entries {
+							of.Batches[i].AddADVEntry(advEntry)
+						}
+					} else {
+						entries := batch.GetEntries()
+						for _, entry := range entries {
+							// Reset TraceNumber
+							entry.TraceNumber = ""
+							of.Batches[i].AddEntry(entry)
+						}
+					}
+					_ = of.Batches[i].Create()
 				}
 			}
 		}
 	}
 
-	// Reset TraceNumber
-	for _, ofBatch := range of.Batches {
-		for _, ofEntry := range ofBatch.GetEntries() {
-			ofEntry.TraceNumber = ""
+	if f.IATBatches != nil {
+		// Slice of IATBatchHeaders
+		sIATBh := make([]string, 0)
+		for _, iatB := range f.IATBatches {
+			IATBh := iatB.GetHeader().String()
+			sIATBh = append(sIATBh, IATBh[:87])
+		}
+		// Remove duplicate IATBatchHeader entries
+		sIATBh = removeDuplicateBatchHeaders(sIATBh)
+		// Add new IATBatches for flattened file
+		for _, record := range sIATBh {
+
+			IATBh := flattenIATBatchHeaderParse(record)
+
+			b := NewIATBatch(IATBh)
+			of.AddIATBatch(b)
+		}
+		for _, iatBatch := range f.IATBatches {
+			fbh := iatBatch.GetHeader().String()[:87]
+			// Add entries for IATBatches
+			for i, ofBatch := range of.IATBatches {
+				if strings.EqualFold(fbh, ofBatch.GetHeader().String()[:87]) {
+					iatEntries := iatBatch.GetEntries()
+					for _, iatEntry := range iatEntries {
+						// reset TraceNumber
+						iatEntry.TraceNumber = ""
+						of.IATBatches[i].AddEntry(iatEntry)
+					}
+				}
+				_ = of.IATBatches[i].Create()
```

Comment:
Should we check this error? It likely won't error because all we change is the TraceNumber, right?

Reply:
and the original file is validated